torch/_inductor/decomposition.py

```diff
@@ -207,6 +207,13 @@ def mm(self, input2):
 
 @register_decomposition([aten.cat.default])
 def cat(tensors, dim=0):
+    def non_empty_tensor(x):
+        # special case for cat'ing with an empty tensor -
+        # just drop the 'empty' inputs so they don't confuse the logic below.
+        return len(x.shape) > 1 or x.shape[0] > 0
+
+    tensors = list(filter(non_empty_tensor, tensors))
```

Comment:
This doesn't do anything, you need to call `aten.cat` if the lists has changed, i.e.:
```
if len(filtered_tensors) != len(tensors):
    return aten.cat(filtered_tensors, dim=dim)
```

And you'll probably need a special case for the empty list.

Reply:
ahh you're right. but why does my test pass :/